src/web/components/structure/LanguageSwitch.jsx

```diff
@@ -34,25 +36,44 @@ const LanguageSwitch = () => {
   };
 
   const handleLanguageChange = async () => {
+    if (isChangingLanguage) {
+      return;
+    }
+
     try {
-      changeLanguage(nextLanguage);
+      const newLanguage = nextLanguage;
+      setIsChangingLanguage(true);
+
+      await new Promise((resolve, reject) => {
+        i18n.changeLanguage(newLanguage, err => {
+          if (err) {
+            reject(err instanceof Error ? err : new Error(err));
+          } else {
+            resolve();
+          }
+        });
+      });
```

Comment:
I don't think this should be necessary. But ff we really need to wait until the change is done, I prefer to change `setLanguage` of `useLanguage` into a promise.
```suggestion
```